### PR TITLE
HHH-13073 - AnyMetaDef.metaType should default to "string"

### PR DESCRIPTION
**Pull request by johnlinp**
https://hibernate.atlassian.net/browse/HHH-13073

### CODE REVIEW

**Comment by gsmet** [comment]
@gbadner what's your take on this?

File: hibernate-core/src/main/java/org/hibernate/annotations/AnyMetaDef.java [hunk 1]
```diff
@@ -37,7 +37,7 @@
 	 * Names the discriminator Hibernate Type for this Any/ManyToAny mapping.  The default is to use
 	 * {@link org.hibernate.type.StringType}
 	 */
-	String metaType();
+	String metaType() default "string";
```
Comment (gbadner) [inline hibernate-core/src/main/java/org/hibernate/annotations/AnyMetaDef.java]: @johnlinp, please add a test case that exercises this default.

File: hibernate-core/src/main/resources/org/hibernate/hibernate-mapping-4.0.xsd [hunk 2]
```diff
@@ -395,7 +395,7 @@ arbitrary number of queries, and import declarations of arbitrary classes.
     <xs:attribute name="index" type="xs:string"/> <!-- include the columns spanned by this association in an index -->
     <xs:attribute name="insert" default="true" type="xs:boolean"/>
     <xs:attribute name="lazy" default="false" type="xs:boolean"/>
-    <xs:attribute name="meta-type" type="xs:string"/> <!--- default: Hibernate.STRING -->
+    <xs:attribute name="meta-type" default="string" type="xs:string"/> <!--- default: Hibernate.STRING -->
```
Comment (gbadner) [inline hibernate-core/src/main/resources/org/hibernate/hibernate-mapping-4.0.xsd]: I'm not sure the change above is actually needed. Please give it a try without your fix to find out. If it's not necessary, then please add a commit with your test and back out of this change and the change to hibernate-mapping-3.0.dtd.

File: hibernate-core/src/main/java/org/hibernate/annotations/AnyMetaDef.java (hunk 1 above)
Comment (johnlinp) [reply hibernate-core/src/main/java/org/hibernate/annotations/AnyMetaDef.java]: OK, I'll do it.

File: hibernate-core/src/main/resources/org/hibernate/hibernate-mapping-4.0.xsd (hunk 2 above)
Comment (johnlinp) [reply hibernate-core/src/main/resources/org/hibernate/hibernate-mapping-4.0.xsd]: Yes, my change to 4.0.xsd is necessary as well. I'll add unit tests to it.

Comment (johnlinp) [reply hibernate-core/src/main/resources/org/hibernate/hibernate-mapping-4.0.xsd]: Sorry, I made a stupid mistake. My change to 4.0.xsd is wrong. I'll figure it out.

Comment (johnlinp) [reply hibernate-core/src/main/resources/org/hibernate/hibernate-mapping-4.0.xsd]: I found that modifying `hibernate-mapping-4.0.xsd` has no effect on the behavior. Instead, modifying `legacy-mapping-4.0.xsd` will actually fix this issue.

File: hibernate-core/src/main/java/org/hibernate/annotations/AnyMetaDef.java (hunk 1 above)
Comment (johnlinp) [reply hibernate-core/src/main/java/org/hibernate/annotations/AnyMetaDef.java]: I removed the property `metaType = "string"` in the existing unit tests to exercise my change.

**Comment by johnlinp** [comment]
@gbadner I updated my fix and added unit tests for my fix. Please take a look at it. Thank you.

**Comment by johnlinp** [comment]
Hi @gbadner, do you have any comment on my patch? Thanks!

**Comment by johnlinp** [comment]
Hi @gsmet, any updates?

**Comment by vladmihalcea** [comment]
@johnlinp superseded by [this PR](https://github.com/hibernate/hibernate-orm/pull/2687).

**Pull request closed without merging**